File: Puzzle/src/imgs/imageCutter.py
from PIL import Image

def cut_image(input_path, output_folder, rows, columns):
    # Open the image
    original_image = Image.open(input_path)

    # Get the size of the original image
    width, height = original_image.size

    # Calculate the width and height of each part
    part_width = width // columns
    part_height = height // rows

    # Loop through each row and column to cut the image
    for row in range(rows):
        for col in range(columns):
            left = col * part_width
            top = row * part_height
            right = (col + 1) * part_width
            bottom = (row + 1) * part_height

            # Crop the image
            part = original_image.crop((left, top, right, bottom))

            # Save the part to the output folder
            part.save(f"{output_folder}/part_{row}_{col}.png")

if __name__ == "__main__":
    input_image_path = "Puzzle/src/imgs/gtav.jpg"
    output_folder_path = "Puzzle/src/imgs/imgs-cuted"
    num_rows = 5
    num_columns = 5

    cut_image(input_image_path, output_folder_path, num_rows, num_columns)

File: Puzzle/src/imgs/imageCutter.js
function cutImage(inputImagePath, outputFolderPath, numRows, numColumns) {
    const originalImage = new Image();
    originalImage.src = inputImagePath;

    originalImage.onload = function () {
        const imageWidth = originalImage.width;
        const imageHeight = originalImage.height;

        const canvas = document.createElement('canvas');
        canvas.width = imageWidth;
        canvas.height = imageHeight;
        const context = canvas.getContext('2d');
        context.drawImage(originalImage, 0, 0);

        const partWidth = Math.floor(imageWidth / numColumns);
        const partHeight = Math.floor(imageHeight / numRows);

        for (let row = 0; row < numRows; row++) {
            for (let col = 0; col < numColumns; col++) {
                const left = col * partWidth;
                const top = row * partHeight;
                const right = (col + 1) * partWidth;
                const bottom = (row + 1) * partHeight;

                const imageData = context.getImageData(left, top, partWidth, partHeight);

                const canvasPart = document.createElement('canvas');
                canvasPart.width = partWidth;
                canvasPart.height = partHeight;
                const contextPart = canvasPart.getContext('2d');
                contextPart.putImageData(imageData, 0, 0);

                const imagePart = new Image();
                imagePart.src = canvasPart.toDataURL('image/png');

                // Aquí puedes manejar o guardar las partes como desees
                document.body.appendChild(imagePart);
            }
        }
    };
}

// Llamada a la función con tus rutas y dimensiones deseadas
cutImage('gtav.jpg', 'imgs-cuted', 5, 5);